corehq/apps/app_manager/models.py

```diff
@@ -984,6 +992,17 @@ def all_other_forms_require_a_case(self):
         m = self.get_module()
         return all([form.requires == 'case' for form in m.get_forms() if form.id != self.id])
 
+    def session_var_for_action(self, action_type, subcase_index=None):
+        module_case_type = self.get_module().case_type
+        if action_type == 'open_case':
+            return 'case_id_new_{}_0'.format(module_case_type)
+        if action_type == 'subcase':
+            opens_case = 'open_case' in self.active_actions()
+            subcase_type = self.actions.subcases[subcase_index].case_type
+            if opens_case and subcase_type == module_case_type:
+                subcase_index += 1  # make sure we don't get two ids: case_id_new_type_0
+            return 'case_id_new_{}_{}'.format(subcase_type, subcase_index)
```

Comment:
Note to self. Think this might be better if the index is just based on the number of actions as opposed to the case_type